drivers/ra/fsp/inc/api/r_usb_device_api.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2020 - 2024 Renesas Electronics Corporation and/or its affiliates
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef R_USB_DEVICE_API_H
+#define R_USB_DEVICE_API_H
+
+/***********************************************************************************************************************
+ * Includes
+ **********************************************************************************************************************/
+
+/* Includes board and MCU related header files. */
+#include "bsp_api.h"
+
+/* Common macro for FSP header files. There is also a corresponding FSP_FOOTER macro at the end of this file. */
+FSP_HEADER
+
+/**********************************************************************************************************************
+ * Macro definitions
+ **********************************************************************************************************************/
+
+/**********************************************************************************************************************
+ * Typedef definitions
+ **********************************************************************************************************************/
+
+/** USB setup packet */
+typedef struct __PACKED st_usbd_setup
+{
+    uint16_t request_type;
+    uint16_t request_value;
+    uint16_t request_index;
+    uint16_t request_length;
```

Comment:
At this time, I think it not really needed to add doxygen comment for all define in this PR. Because we do not have any build system to generate FSP documentation, we cannot confirm whether the documentation can be generated correctly. So, at first, I think it is better to add minimal comment for typedef and API.

Reply:
Btw, the definition in this file is standard and have document well by USB2.0 specs. So, I think it not hard for other maintainer to look up information for these define and add doxygen document for it later.